### PR TITLE
#7560: Add SD tests to FD nightly with legacy pass

### PR DESCRIPTION
**Pull request by mtatsumiTT**
it seems running the legacy pass of cross-attention module for certain input shapes works around the ND hang, I added a pytest option to selectively run the pass instead of `time_sharded_attention` as a work-around for now.

fyi @AleksKnezevic

### CODE REVIEW

File: tests/ttnn/conftest.py [hunk 1]
```diff
@@ -48,3 +48,21 @@ def pre_and_post(request):
 
     ttnn.tracer.disable_tracing()
     ttnn.CONFIG = original_config
+
+
+def pytest_addoption(parser):
```
Comment (tt-rkim) [inline tests/ttnn/conftest.py:53]: Did you want to move this up to top-level conftest.py?

Comment (mtatsumiTT) [reply tests/ttnn/conftest.py:53]: ah yes, I didn't notice there is another conftest.py. I'll move it to there

**Review by arakhmati — APPROVED** [review]

File: tests/scripts/nightly/run_wh_b0_only.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ fi
 echo "Running nightly tests for WH B0 only"
 
 env pytest tests/ttnn/integration_tests/unet                # -> failing: issue #7556
+env WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest tests/ttnn/integration_tests/stable_diffusion -k 512 --option legacy
```
Comment (tt-rkim) [inline tests/scripts/nightly/run_wh_b0_only.sh:13]: Can you not use `-k`?

Comment (mtatsumiTT) [reply tests/scripts/nightly/run_wh_b0_only.sh:13]: sure, will remove it

**Pull request closed without merging**